stream/src/main/scala/org/apache/pekko/stream/impl/fusing/StreamOfStreams.scala

```diff
@@ -614,14 +607,31 @@ import pekko.util.ccompat.JavaConverters._
 
       override def onDownstreamFinish(cause: Throwable): Unit = {
         substreamCancelled = true
-        if (isClosed(in) || propagateSubstreamCancel(cause)) {
-          cancelStage(cause)
-        } else {
-          // Start draining
-          if (!hasBeenPulled(in)) pull(in)
+        decider(cause) match {
```

Comment:
the cause can be `NoMoreElementsNeeded | StageWasCompleted` should test that first.

Reply:
Isn't this being tested in a different way later down with `isClosed`/`hasBeenPulled`? Also I had a look at parts of the Pekko codebase that implements `Supervision.restart` and I don't see `NoMoreElementsNeeded | StageWasCompleted` being tested in this way.

This doesn't mean its wrong, but if thats the case then testing `NoMoreElementsNeeded | StageWasCompleted` would be unique to `SubFlow` and I am not sure why?

Reply:
NoMoreElementsNeeded | StageWasCompleted indicate the downstream cancel without exception, then that should not go through the supervision decider ? @raboof @jrudolph may have better understanding than me , ping~